File: src/main/java/toby/command/commands/ChCommand.java
package toby.command.commands;

import net.dv8tion.jda.api.entities.Message;
import net.dv8tion.jda.api.entities.TextChannel;
import toby.command.CommandContext;
import toby.command.ICommand;
import toby.jpa.dto.UserDto;
import toby.jpa.service.IConfigService;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class ChCommand implements ICommand {

    private final IConfigService configService;

    public ChCommand(IConfigService configService) {
        this.configService = configService;
    }

    @Override
    public void handle(CommandContext ctx, String prefix, UserDto requestingUserDto) {
        final TextChannel channel = ctx.getChannel();
        final Message message = ctx.getMessage();

        String newMessage = Arrays.stream(message.getContentRaw().split(" ")).map(s -> {
            if (s.equalsIgnoreCase(String.format("%sch", prefix))) {
                return "";
            } else {
                int vowelIndex = 0;
                for (int i = 0; i < s.length(); i++) {
                    char c = s.charAt(i);
                    if (isVowel(String.valueOf(c))) {
                        vowelIndex = i;
                        break;
                    }
                }
                if (s.substring(vowelIndex).toLowerCase().startsWith("ink")) {
                    return "gamerword";
                } else return "ch" + s.substring(vowelIndex).toLowerCase();
            }
        }).collect(Collectors.joining(" "));

        channel.sendMessage("Oh! I think you mean: '" + newMessage.stripLeading() + "'").queue();
    }


    @Override
    public String getName() {
        return "ch";
    }

    @Override
    public String getHelp(String prefix) {
        return "Allow me to translate whatever you type.\n" +
                String.format("Usage: `%sch example message here`", prefix);
    }

    private boolean isVowel(String s) {
        List<String> vowels = List.of("a", "e", "i", "o", "u");
        return vowels.contains(s.toLowerCase());
    }
}

File: discord-bot/src/main/kotlin/bot/toby/command/commands/misc/ChCommand.kt
package bot.toby.command.commands.misc

import bot.toby.command.CommandContext
import bot.toby.command.ICommand.Companion.invokeDeleteOnMessageResponse
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.OptionData
import org.springframework.stereotype.Component
import java.util.*

@Component
class ChCommand : IMiscCommand {
    override val name: String = "ch"
    override val description: String = "Allow me to 'ch' whatever you type."
    private val MESSAGE = "message"

    override fun handle(ctx: CommandContext, requestingUserDto: bot.database.dto.UserDto, deleteDelay: Int?) {
        val event = ctx.event
        event.deferReply().queue()

        val newMessage = (event.getOption(MESSAGE)?.asString ?: "").split(" ").joinToString(" ") { word ->
            val vowelIndex = word.indexOfFirst { isVowel(it.toString()) }
            if (word.substring(vowelIndex).lowercase(Locale.getDefault()).startsWith("ink")) {
                "gamerword"
            } else {
                "ch" + word.substring(vowelIndex).lowercase(Locale.getDefault())
            }
        }

        event.hook.sendMessage("Oh! I think you mean: '$newMessage'").queue(invokeDeleteOnMessageResponse(deleteDelay ?: 0))
    }

    override val optionData: List<OptionData>
        get() = listOf(OptionData(OptionType.STRING, MESSAGE, "Message to 'Ch'", true))

    private fun isVowel(s: String): Boolean {
        val vowels = listOf("a", "e", "i", "o", "u")
        return vowels.contains(s.lowercase(Locale.getDefault()))
    }
}